### PR TITLE
fix NPE in IndexGeneratorJob

### PR DESCRIPTION
**Pull request by himanshug**
in hadoop 2.8 "mapred.job.tracker" property is not present by default and leads to a NPE.

### CODE REVIEW

**Comment by b-slim** [comment]
how about check for `mapreduce.jobtracker.address ` as well ?

**Comment by himanshug** [comment]
@b-slim ok, added that one as well.

in the long term, I think that if conditional is probably not needed anymore. it was added there to workaround a hadoop issue where it wouldn't propagate the real number of reducers but just 1 all the time when local job tracker was used.

File: indexing-hadoop/src/main/java/io/druid/indexer/IndexGeneratorJob.java [hunk 1]
```diff
@@ -454,7 +454,7 @@ public int getPartition(BytesWritable bytesWritable, Writable value, int numPart
       final ByteBuffer bytes = ByteBuffer.wrap(bytesWritable.getBytes());
       bytes.position(4); // Skip length added by SortableBytes
       int shardNum = bytes.getInt();
-      if (config.get("mapred.job.tracker").equals("local")) {
+      if ("local".equals(config.get("mapreduce.job.tracker")) || "local".equals(config.get("mapred.job.tracker"))) {
```
Comment (nishantmonu51) [inline indexing-hadoop/src/main/java/io/druid/indexer/IndexGeneratorJob.java]: mapreduce.job.tracker -> mapreduce.jobtracker.address

**Comment by b-slim** [comment]
Ci job failed with `The job exceeded the maximum time limit for jobs, and has been terminated.`

File: indexing-hadoop/src/main/java/io/druid/indexer/IndexGeneratorJob.java (hunk 1 above)
Comment (himanshug) [inline indexing-hadoop/src/main/java/io/druid/indexer/IndexGeneratorJob.java]: thanks :) , should've copy/pasted

**Comment by fjy** [comment]
👍

**Pull request merged**